suite-native/device/src/hooks/useDeviceAuthenticityCheck.tsx

```diff
@@ -0,0 +1,199 @@
+import { useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { useNavigation } from '@react-navigation/native';
+
+import {
+    CheckDeviceAuthenticityThunkResult,
+    deviceAuthenticityActions,
+} from '@suite-common/device-authenticity';
+import { selectSelectedDevice } from '@suite-common/wallet-core';
+import { DeviceAuthenticityCheckResult, EventType, analytics } from '@suite-native/analytics';
+import { requestPrioritizedDeviceAccess } from '@suite-native/device-mutex';
+import { FeatureFlag, useFeatureFlag } from '@suite-native/feature-flags';
+import { Translation } from '@suite-native/intl';
+import { captureSentryMessage, withSentryScope } from '@suite-native/sentry';
+import { useToast } from '@suite-native/toasts';
+import TrezorConnect from '@trezor/connect';
+
+export const useDeviceAuthenticityCheck = () => {
+    const navigation = useNavigation();
+    const dispatch = useDispatch();
+    const { showToast } = useToast();
+    const allowDebugKeys = useFeatureFlag(FeatureFlag.IsDebugKeysAllowed);
+
+    const device = useSelector(selectSelectedDevice);
+    const isDeviceBootloaderUnlocked = !!device && !device?.features?.bootloader_locked;
+    const reportCheckResult = useCallback(
+        (
+            result: DeviceAuthenticityCheckResult,
+            error?: string,
+            payload?: CheckDeviceAuthenticityThunkResult,
+        ) => {
+            analytics.report({
+                type: EventType.DeviceSettingsAuthenticityCheck,
+                payload: { result },
+            });
+            if (['compromised', 'configExpired', 'failed'].includes(result)) {
```

Comment:
It always makes me happy when I can sell my TS utils :grin: 
Then you can remove `as 'compromised' | 'configExpired' | 'failed'` a few lines below, because it narrows down the type :slightly_smiling_face: 
```suggestion
            if (isArrayMember(result, ['compromised', 'configExpired', 'failed'])) {
```

Reply:
Wow, so much better, thanks for letting me know about this util!

Reply:
[`4e56e00` (#18766)](https://github.com/trezor/trezor-suite/pull/18766/commits/4e56e00823dce0f82623e4830a3fdf32b29d6d25)